ext2/ext2.c

```diff
@@ -5080,12 +5085,22 @@ static int _ext2_getdents64(
     if (!_ext2_valid(ext2) || !_file_valid(file) || !dirp)
         ERAISE(-EINVAL);
 
-    if (!(file->open_flags & O_DIRECTORY))
+    if (!S_ISDIR(file->inode.i_mode))
         ERAISE(-ENOTDIR);
 
     if (count == 0)
         goto done;
 
+    if (file->dir.data == NULL)
```

Comment:
Please add comments on when this would happen.

Reply:
Added comment.